config/types/etcd.go

```diff
@@ -32,7 +33,9 @@ var (
 )
 
 // Options can be the options for any Etcd version
-type Options interface{}
+type Options interface {
```

Comment:
I didn't dig into the implementation of the validation logic, but the code wasn't called without this change. I can spend some time digging into the validation logic if you'd prefer me to try to fix it in there, but I have a feeling it's due to it not being able to know what functions are available to call on an empty interface.